### ISSUE TITLE
Rendering is stuck on Windows for a simple doc 

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#5942]
```markdown
---
title: Test including figure in docx
format: html
keep-md: true
---

![This is an elephant.](elephant.png){#fig-elephant width=650}

```

(`elephant.png` is from our website, but any image will cause issue)

`quarto render` on this doc, and process is stuck at 
````powershell
> quarto render test2.qmd
Quarto version: 99.9.9
Loaded deno-dom-native
Windows: Using code page 1252
pandoc --verbose --trace
  to: html
  output-file: test2.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Test including figure in docx

[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\template.patched from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\template.patched
[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\metadata.html from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\metadata.html
[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\styles.html from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\styles.html
[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-block.html from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-block.html
[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-metadata.html from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-metadata.html
[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-block.html from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-block.html
[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-metadata.html from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\title-metadata.html
[INFO] Loaded C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\toc.html from C:\Users\chris\AppData\Local\Temp\quarto-session378b8c77\17dd0693\0dc0abb2\toc.html
[trace] Parsed [Plain [Str "long"]] at line 1
[trace] Parsed [Plain [Str "en"]] at line 1
[trace] Parsed [Plain [Str "true"]] at line 1
[trace] Parsed [Plain [Str "99.9.9"]] at line 1
[trace] Parsed [Plain [Str "Test",Space,Str "including",Space,Str "figure", at line 1
[trace] Parsed [Plain [Str "Table",Space,Str "of",Space,Str "contents"]] at line 1
[trace] Parsed [] at line 5
[trace] Parsed [Figure ("fig-elephant",[],[]) (Caption Nothing [Plain [Str  at line 8
[INFO] Running filter C:/Users/chris/Documents/DEV_R/quarto-cli/src/resources/filters/main.lua
[INFO] Completed filter C:/Users/chris/Documents/DEV_R/quarto-cli/src/resources/filters/main.lua in 46 ms
^CTerminer le programme de commandes (O/N) ? O
````

Took me sometime to find as I tried a few things. 

* Trying to use the tracing feature gave me the hint of infinite loop as the trace created is more than  2 Gb (!!) 

* I ended up running a git bisect to find where it happens, to understand why. 

Culprit is https://github.com/quarto-dev/quarto-cli/commit/553fec5f3c421a8c0e9a533ef47d76c6f002e437#diff-a86a3d2ddd41304881ed9f8418b10f3e960315213da5cf51b76a687c88ef1e50R53 (👋 @dragonstyle ) 

Running a debug mode in VSCODE showed me the difference 

in here
https://github.com/quarto-dev/quarto-cli/blob/1f3581304d2d1112ae22cd970e7c176d8be15600/src/core/path.ts#L275-L276

we were looking for `expandGlobs("elephant.png")` but after the commit, we are looking for `expandGlobs("**\elephant.png")`

The difference is small in the commit but the strict mode was removed 
https://github.com/quarto-dev/quarto-cli/blob/9402ca9ad15b8c693fcc2c7691df3becb3d3631e/src/command/render/resources.ts#L54-L56

So this mode was setting smartGlob to FALSE
https://github.com/quarto-dev/quarto-cli/blob/1f3581304d2d1112ae22cd970e7c176d8be15600/src/core/path.ts#L209-L226

but now, it is set to TRUE and we end up going into any `if(smartGlob)` 

Hope it makes it clear. I can detail more and show what happens. 

@dragonstyle any insight why remove the strict mode and then change the smartGlob value ? 

It does not explain the infinite loop but it is related to 
```
expandGlobs("**\elephant.png")
```
on Windows

**Comment by dragonstyle** [comment]
Hmm.. I'm thinking that there must be something about the folder that you're rendering within? I can't seem to reproduce the issue at all on Windows or MacOS, but I'm thinking maybe the recursive expansion of the glob is causing the issue because of something about the environment / path on your machine. Here's what my dirs look like:

```
PS C:\Users\User\test> ls


    Directory: C:\Users\User\test


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         6/15/2023   2:09 PM                test_files
d-----         6/15/2023   2:02 PM                _extensions
-a----         6/15/2023   2:05 PM         211643 elephant.jpg
-a----         6/15/2023   2:07 PM         111228 elephant.png
-a----         6/15/2023   2:09 PM         121631 test.docx
-a----         6/15/2023   2:07 PM          12911 test.html
-a----         6/15/2023   2:09 PM            150 test.md
-a----         6/15/2023   2:07 PM            147 test.qmd


PS C:\Users\User\test>
```

and qmd:

```
---
title: Test including figure in docx
format: html
keep-md: true
---

![This is an elephant.](elephant.png){#fig-elephant width=650}
```

Just to confirm I'm reproing correctly...

**Comment by cderv** [comment]
That is really helpful ! I am definitely not in a clean folder ! Though something has changed with that commit, so I'll look into this with this in mind. Thank you !

**Comment by dragonstyle** [comment]
We definitely did change things (you're right that we're now smart globbing which in this case I'm expecting mean we're doing a recursive search where we previously were not). I have a slight suspicion that we're somehow searching a very unexpected folder path (e.g. we're 'escaping' the project directory or something....)

**Comment by dragonstyle** [comment]
Yeah I agree I think it is surprising - I think trying to pinpoint where the glob is being expanded and being sure it really is in the document directory as the base path. An example fear of mine would be somehow `..\..\` being prepended to the path or base path, resulting in a much larger search than expected / healthy.

**Comment by cderv** [comment]
@dragonstyle I think I am missing something about how this should work. Current situation with the smart glob `**/elephant.png` is that it will find all the `elephant.png` file that exists in subdirectories. It happens I am using a filename we use in several Quarto projects and demo so there are a few. It is taking a long time, but finding 38 hits - as I am using my main Home folder to run this. 

It seems that those 38 files are considered as resource Files that goes with the document. Maybe this is what also adds to the timing for the rendering. 

Anyhow, I think I don't understand why when we have  such a simple qmd content that is inserting an image
````markdown
![This is an elephant.](elephant.png){#fig-elephant width=650}
````

we need to look into subfolders also with `**/elephant.png`, using globstart. I am probably missing something related to another feature. Could be worth to think about it together. 

It could be a edge case, but something does not feel right to me. Also, I am using this workflow (`quarto render` a simple qmd from a desktop or root `~` quite often) - Maybe not the only one among windows users. 🤷‍♂️ Definitely the only one to have 38 files as results of `expandGlobs` though 😅

**Comment by dragonstyle** [comment]
Yeah, I'm not thinking that this is desirable behavior at all - was just pointing to the likely source of the bug. The change was originally made because resources were not being properly watched during preview, b/c the paths were not being treated as 'smart', but that initial attempt to fix the issue here:

https://github.com/quarto-dev/quarto-cli/issues/5611

Resulted in a new issue, since the path`../` is started with a `.` character:

https://github.com/quarto-dev/quarto-cli/issues/5774

The result of all of these changes is that we go from strict when interpreting resources to automatic (which shifts us into recursive searches in places we wouldn't otherwise be searching recursively). 

I think the core question is really just a design question of whether resources should be smart globs or not. If they are not, then we should probably just revert the change that to force it back into strict mode. 

The other question that occurs to me is why we're treating `elephant.png` as a glob at all - I would expect that globs are not allowed as the src of figures, and so we perhaps should be resolving that into a resource file but are somehow instead treating it as a resource glob. This is something I'll try to take a look at.

**Comment by cderv** [comment]
> The other question that occurs to me is why we're treating elephant.png as a glob at all - I would expect that globs are not allowed as the src of figures, and so we perhaps should be resolving that into a resource file but are somehow instead treating it as a resource glob. This is something I'll try to take a look at.

Regarding this, during my debug I think the `elephant.png` file was correctly found once as `renderedFile.resourceFiles` but then transformed as a glob and search as `fileResourceFiles`
and we make one from from at
https://github.com/quarto-dev/quarto-cli/blob/eb2b68fa040a0b95c39e85d6d1851ea6793a1343/src/command/render/resources.ts#L93-L99

Still need to be confirmed by another debug run... Just sharing in case it can help

**Comment by cderv** [comment]
I believe this is solved with 1.4 now. Not sure why, but I can't reproduce.

I'll close and reopen if I encounter it again.
